src/ompl/geometric/planners/rrt/src/RRTConnect.cpp

```diff
@@ -244,9 +248,36 @@ ompl::base::PlannerStatus ompl::geometric::RRTConnect::solve(const base::Planner
 
     while (!ptc)
     {
+        // Tree balancing heuristic: preferentially extend the smaller tree to keep trees balanced
+        // This helps prevent one tree from dominating exploration (recommended by Kuffner)
+        if (balanceTreeSizes_ && tStart_->size() > 0 && tGoal_->size() > 0)
+        {
+            std::size_t startSize = tStart_->size();
+            std::size_t goalSize = tGoal_->size();
+            double sizeRatio = std::abs(static_cast<double>(startSize) - static_cast<double>(goalSize)) / 
+                              static_cast<double>(std::max(startSize, goalSize));
+            
+            // Only swap if size difference is acceptable (ratio < threshold)
+            // Otherwise, keep extending the smaller tree
+            if (sizeRatio >= maxTreeSizeRatio_)
```

Comment:
this bunch of ifs is unnecessarily complicated

cf. https://github.com/KavrakiLab/vamp/blob/main/src/impl/vamp/planning/rrtc.hh#L104